src/openms_gui/source/VISUAL/ANNOTATION/Annotation1DPeakItem.cpp

```diff
@@ -196,20 +196,78 @@ namespace OpenMS
       painter.setPen(Qt::DashLine);
       if (!found_intersection) // no intersection with bounding box of text -> normal drawing
       {
-        painter.drawLine(peak_position_widget, position_widget);
-        painter.drawLine(peak_position_widget, position_widget);
+        painter.drawLine(peak_position_widget, peak_position_widget + QPoint(0, vertical_shift));
+        painter.drawLine(peak_position_widget + QPoint(0, vertical_shift), position_widget);
       }
       else
       {
-        painter.drawLine(peak_position_widget, *closest_ip);
-        painter.drawLine(peak_position_widget, *closest_ip);
+        QPoint vertical_end = QPoint(peak_position_widget.x(), position_widget.y() + 20);
+
+        if (vertical_end.y() < peak_position_widget.y()) // label is above peak
+        {
+          painter.drawLine(peak_position_widget, vertical_end);
+          painter.drawLine(vertical_end, position_widget);   
+        }
+        else // label is below peak
+        {
+          painter.drawLine(peak_position_widget, *closest_ip);
+          painter.drawLine(peak_position_widget, *closest_ip);       
+        }        
       }
       painter.restore();
       delete(ip);
       delete(closest_ip);
     }
 
-    painter.drawText(bounding_box_, Qt::AlignCenter, text_);
+    // some pretty printing
+    QString text = text_;
+    if (!text.contains("<\\")) // don't process HTML strings again
+    {
+      // extract ion index
+      QRegExp reg_exp("[a|b|c|x|y|z](\\d+)");
+      int match_pos = reg_exp.indexIn(text);
+      if (match_pos == 0)
+      {
+        QString index_str = reg_exp.cap(1);
+
+        // put sub html tag around number
+        text = text[0]
+              + QString("<sub>") + index_str + QString("</sub>")
+              + text.right(text.size() - match_pos - index_str.size() - 1);
+      }
+
+      // charge
+      text.replace(QRegExp("3\\+$"), "<sup>3+</sup>");
+      text.replace(QRegExp("\\+\\+$"), "<sup>2+</sup>");
+      text.replace(QRegExp("\\+$"), "");
+      text.replace(QRegExp("3\\-$"), "<sup>3-</sup>");
+      text.replace(QRegExp("\\-\\-$"), "<sup>2-</sup>");
+      text.replace(QRegExp("\\-$"), "");
+
+      // common losses
+      text.replace("H2O","H<sub>2</sub>O");
+      text.replace("H2O1","H<sub>2</sub>O");
```

Comment:
The order is wrong, `H2O` is replaced first, and then `H<sub>2</sub>O1` will not be seen as `H2O1` and the `1` stays. This happens to my labels.
For `NH3` the two versions are sufficiently different, so this does not happen there.

Reply:
aaah great catch! Thanks!